packages/next/server/next-server.ts

```diff
@@ -1641,7 +1646,10 @@ export default class Server {
       } else if (isRedirect) {
         value = { kind: 'REDIRECT', props: pageData }
       } else {
-        value = { kind: 'PAGE', html: html!, pageData }
```

Comment:
This is the underlying bug that this PR addresses: It's valid for `html` to be `null` in some cases, but not for a cached page. The `!` just makes the typechecking wrong.